pkg/apis/core/v1beta1/constants/types_constants.go

```diff
@@ -499,6 +499,10 @@ const (
 	// should not be deleted if the corresponding `Project` gets deleted. Please note that all project related labels
 	// from the namespace will be removed when the project is being deleted.
 	NamespaceKeepAfterProjectDeletion = "namespace.gardener.cloud/keep-after-project-deletion"
+	// NamespaceCreatedByProject is a constant for annotation on a `Namespace` resource that states that it
+	// was created by the project because either the user didn't specify the namespace in the project `spec.Namespace`
+	// or the specified namespace was not present.
+	NamespaceCreatedByProject = "namespace.gardener.cloud/created-by-project"
```

Comment:
I would rather vote to rename the annotation to `namespace.gardener.cloud/created-by-project-controller` to clearly indicate the "author" of the create operation.

```suggestion
	NamespaceCreatedByProjectController = "namespace.gardener.cloud/created-by-project-controller"
```